lib/compress/index.js

```diff
@@ -1598,6 +1598,38 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every((branch, i) =>
+            (body.length - 1) === i // last branch
+            || branch instanceof AST_Default // or default:
+            || !branch.expression.has_side_effects(compressor) // or no_side_effects
+            && (
+                branch.body.length === 0 // or empty
+                || is_break(branch.body[branch.body.length - 1], compressor) // or has a break
+                || is_return(branch.body[branch.body.length - 1]) // or has a return
+            )
+    )) {
```

Comment:
We might be able to improve this further. It's safe to perform reorganizations up to the first branch expression with a side-effect. Then up to the next side-effect, then the next, etc. So no need to check everything is side-effect free. The only thing we need to watch out for are non-aborting branches, since we cannot move the following branch from its location.

Reply:
Yeah I know, it started out like that, but testing was taking me too much time, and I was writing this code at 03:00 Israel time, so I had to draw the line somewhere 😅

I think the gains are substantial even with this simple approach. 
I would rather do the additional improvement in another PR if that’s fine with you.

Reply:
> I would rather do the additional improvement in another PR if that’s fine with you.

Sounds good to me!

Reply:
Yah, split PRs is fine. Mine got toooo large.